### ISSUE TITLE
How to achieve constant speed rotation?

### ISSUE THREAD

**Issue by pjcozzi** [KhronosGroup/glTF#1056]
At SIGGRAPH, it was brought up that glTF keyframe animations can't achieve constant speed rotation.  This would require slerp or Euler angles.

This could be approximated with more samples - could be an implementation note.

I'm not 100% sure on the slerp comment since the spec does have it: https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#animation-samplerinterpolation

**Comment by emilian0** [comment]
@pjcozzi I was assuming that `LINEAR` interpolation on quaternions had to be interpreted as SLERP. Should we clarify that? Is this ok for everyone?

**Comment by javagl** [comment]
The spec currently says

> `"LINEAR"` The animated values are linearly interpolated between keyframes. When targeting a rotation, spherical linear interpolation (slerp) should be used to interpolate quaternions. 

I think that a slerp should have a "constant speed", but am not sure whether this is the case and thus whether this answers/resolves this issue...

**Comment by emilian0** [comment]
@javagl AFAIK:
t = 0 -> k = (0,0,0,1)
t = 0.5 -> k = (n,1)
t = 1.0 -> k = (0,1)
should rotate around n at constant speed (when interpolating using SLERP).

**Comment by emilian0** [comment]
@pjcozzi @javagl 
I noticed that [this](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/AnimatedCube) glTF2.0 sample (curtesy of Norbert) displays constant rotation.
Just to double check I cracked open the bin file and verified that the animation curve looks like the above (where the axis of rotation n = [0 1 0]).
If there are no further concerns I believe that we can archive this conversation.
